CMakeLists.txt

```diff
@@ -7,13 +7,13 @@
 
 cmake_minimum_required(VERSION 3.5)
```

Comment:
I'm thinking maybe this should be put in a block that only runs if this is the current project. We check policies in at least one more place, I think, and we want the user to be setting this, not us.

Reply:
That's definitely an easy fix and I can make that change, but I was wondering... (and bear with me because I'm by no means a CMake expert) isn't it better to have pybind11 set a maximum `cmake_policy` (which would be scoped to its directories as far as I understand) even when being included by a root project?

What I'm thinking is say the main project has set the CMake _maximum_ version to something more modern that introduced new policies that pybind11 cannot accommodate (for whatever reason). In that case, the `cmake_policy(VERSION XXX)` in pybind11 would still allow all the pybind11 CMake code to run fine, wouldn't it?

The way I see it is that the `cmake_policy` statement in pybind11 is fine as long as all other CMake code in pybind11 is designed to run with that same maximum CMake version and policies. That is currently not the case due to the code in pybind11 checking the CMP0148 policy, which only makes sense when the CMake maximum version is 3.27+.

Please let me know if I'm missing something.

Reply:
It's a project's responsibility when setting a maximum version to make sure it works when using that maximum version. But I'm fine (given the size of the change and the fact I've not really seen it done this way before) to just propagate the policies that we want to monitor and respect. I think that's at least this one & maybe the interprocedural optimization one.

Reply:
Sounds good. Do you want me to make any other changes then? (like maybe setting a CMake variable to control the maximum CMake version in a single spot instead of having to modify several files whenever a change like this is required...)